Model/CheckoutSessionManagement.php

```diff
@@ -593,6 +591,8 @@ private function getTransaction($transactionId, $type = null)
 
         if (count($transactionCollection)) {
             return $transactionCollection->getFirstItem();
+        } else {
+            return null;
```

Comment:
Did you run into a scenario where there was no transaction? The problem with this is that later on on the `$chargeState` switch, the transaction is being used, and this will trigger an exception.

Reply:
No, added just to be explicit. Should we throw an error if is null immediately after calling here?
[https://github.com/BearGroup/amazon-payments-magento-2-plugin/blob/29ad5c41edb398120a3f1539610621146ca2db0e/Model/CheckoutSessionManagement.php#L832](line)

Reply:
Hmmm. Not sure if there's any case where it could go through there without a transaction already present.